lib/types/application/ApplicationFormatter.js

```diff
@@ -20,33 +20,60 @@ class ApplicationFormatter extends AbstractFormatter {
 			};
 
 			return this.readManifest(project).then(function(manifest) {
-				if (!manifest["sap.app"] || !manifest["sap.app"].id) {
+				// check for a proper sap.app/id in manifest.json to determine namespace
+				const appId = manifest["sap.app"] && manifest["sap.app"].id;
+				if (!appId) {
 					log.warn(`No "sap.app" ID configuration found in manifest of project ${project.metadata.name}`);
 					return;
 				}
-				project.metadata.namespace = manifest["sap.app"].id.replace(/\./g, "/");
-			}).catch((err) => {
+				// check app id for placeholder and try to guess namespace from component controller
+				const parts = appId.match(/^\$\{(.*)\}$/);
+				if (parts) {
+					log.warn(`No valid "sap.app" ID found in manifest of project ${project.metadata.name}. Guessing from Component.js...`);
+					return this.readComponentController(project).then((code) => {
+						// use regex to determine namespace from component controller
+						const namespaceParts = code.match(/\.(?:\s+)?extend(?:\s+)?\((?:\s+)?["'](.*)\.Component["'](?:\s+)?,/);
```

Comment:
instead of `(?:\s+)?` use the shorter and more readable `\s*`

Reply:
second argument to extend(name, ...) is optional (although it rarely ever is missing)

Reply:
Overall: why is the namespace read from the Component.js, not from the pom.xml? If the manifest says, ${project.artifactId} is the name, using the name from the extend call is not more than a guess.

Reply:
Reading it from pom.xml was also something I had briefly in mind about but I skipped it due to the fact to align it to the runtime behavior to determine the name of the Component. In addition, I also found out that the template which identified the issue has no pom.xml aside of the manifest.json which is also another issue and must not be happen. So, identifying from the pom.xml should be the better option and the manifest.json may only have such placeholder if a Maven build is aside.